workflow/controller/operator.go

```diff
@@ -240,7 +240,7 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 	woc.taskResultReconciliation()
 
 	// Do artifact GC if all task results are completed.
-	if woc.checkTaskResultsCompleted() {
+	if !woc.checkTaskResultsInProgress() {
```

Comment:
we don't need to check if `Phase.Completed()` for this case?

Reply:
Pretty sure we don't need to because of the strategy check, but I see no harm in doing so.

Reply:
The artifact gc strategy protects us for this case. We could conceivably add this though. It would make things a bit more efficient.

Reply:
plus, if we have common logic between this and the other calls, we might be able to have a common function, which would be nice

Reply:
I'm actually kind of thinking about a common function which checks for the Workflow being completed, plus compares the number of pods that were run to the number of task results that got incorporated into the Workflow Status, which could be called `Workflow.ReconciliationComplete()` or something like that. Does that make any sense?

This method could be called at any place in the code, and it should work.

Reply:
I know in this case you were only concerned about which taskResults were saved at all, as opposed to which ones had actually gone through `podReconciliation()` as well, so maybe the common function wouldn't be used here if it doesn't make sense.

Reply:
(just editing my comment 2 comments back because it wasn't clear what i was talking about)